src/ucs/stats/stats.h

```diff
@@ -19,10 +19,61 @@ BEGIN_C_DECLS
 
 /** @file stats.h */
 
+/* The maximum size of a class name */
+#define MAX_CLS_NAME_SIZE 50
+
+/* The maximum number of UCX aggregate-sum counters */
+#define MAX_NUM_OF_STATS_AGGREGATE_SUM_COUNTERS 32
+
+/* The maximum number of statistics classes */
+#define MAX_NUM_OF_STATS_CLASSES 32
+
+typedef struct tag_ucs_stats_counter {
+    /* Counter class name */
+    const char *class_name;
+
+    /* Counter name */
+    const char *name;
+
+    /* Counter value */
+    uint64_t value;
+
+} ucs_stats_counter_s;
+
+typedef struct tag_ucs_stats_cls_id_to_name {
+    /* Class name */
+    char class_name[MAX_CLS_NAME_SIZE];
+
+    /* Class base offset in the counters aggregate-sum list */
+    unsigned aggregate_sum_class_offset;
+
+} ucs_stats_class_id_to_name_s;
+
+
 void ucs_stats_init();
 void ucs_stats_cleanup();
 void ucs_stats_dump();
 int ucs_stats_is_active();
+
+/**
+ * A UCX statistics API function to returns the aggregate-sum of all counters
+ * from all workers/interfaces for a compact statistics trace.
+ * Used for Score-P UCX plugin to collect a compact trace of the counters.
+ *
+ * @param [out] aggregate_sum_counters      The aggregate-sum of all counters of each class/type.
+ * @param [in]  aggregate_sum_counters_size The number of elements in aggregate_sum_counters.
+ * @param [out] n_counters                  The actual number of counters in the output buffer.
+ *
+ * Complexity: O(n)
+ *
+ * Returns: None.
+ *
+ * Remarks: The function calls a recursive function.
+ */
+void ucs_stats_aggregate_sum(ucs_stats_counter_s *aggregate_sum_counters,
```

Comment:
1. maybe return number of counters?
2. what is thread-safety mode of this function?
3. pls add gtest for it
4. pls try to use shorter names for parameters

Reply:
1. See n_counters_out - This is the actual number of counters in the aggregate-sum. I can also return the total number of counters but I think it will be confusing for the users.
2. It's not thread-safe, thread-safety is on the user's responsibility.
3. I will add a gtest.
4. Will do.

Reply:
3. Fixed.
4. Done.